File: LeetCode/189_RotateArray.py
# https://leetcode.com/problems/rotate-array/
# 2022年07月02日 22:11:01

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # # v1 slow, 2324ms
        # k=k%len(nums)
        # while k>0:
        #     k-=1
        #     nums.insert(0,nums.pop()) 
        
        # v2 slice assignment, 300ms
        k%=len(nums)
        nums[:]=nums[-k:]+nums[:-k] 
        

File: LeetCode/189_RotateArray.go
// https://leetcode.com/problems/rotate-array/
// 2022年07月03日 20:30:03

func rotate(nums []int, k int)  {
    nk:=len(nums)-k%len(nums)
    // copy means copies elements into a certain destination slice from a source slice, is an in-place operation 
    copy(nums,append(nums[nk:],nums[:nk]...))
    // directly assign is not in-place 
    // nums=append(nums[nk:],nums[:nk]...)
}

